sql/create.go

```diff
@@ -42,15 +42,29 @@ func (p *planner) CreateDatabase(n *parser.CreateDatabase) (planNode, error) {
 		}
 		return nil, fmt.Errorf("database \"%s\" already exists", n.Name)
 	}
+
 	ir, err := p.db.Inc(keys.DescIDGenerator, 1)
 	if err != nil {
 		return nil, err
 	}
-	nsID := uint32(ir.ValueInt() - 1)
+
+	desc := makeDatabaseDesc(n)
+	desc.ID = uint32(ir.ValueInt() - 1)
+
+	// TODO(pmattis): The error currently returned below is likely going to be
+	// difficult to interpret.
 	// TODO(pmattis): Need to handle if-not-exists here as well.
-	if err := p.db.CPut(nameKey, nsID, nil); err != nil {
+	err = p.db.Txn(func(txn *client.Txn) error {
```

Comment:
This may not convince you, but my feeling is that the shadowing form makes intent clearer: "this error is not meant to escape this scope". "Ugly" is subjective, and so long as we follow the shadowing pattern everywhere we'll get used to it.